### PR TITLE
Haskell http-client-tls stub

### PR DESCRIPTION
**Pull request by oherrala**
This is first Haskell stub to support custom CA bundles given by TryTLS runner. It required modification to Haskell's x509-store package. Pull request about it can be found here: https://github.com/vincenthz/hs-certificate/pull/71

### CODE REVIEW

File: stubs/haskell-http-client-tls/README.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+You can use Docker to build the stub:
+
+```sh
+$ docker build -t test-haskell-client-tls --rm .
+```
+
+and then run it with TryTLS runner:
+
+```sh
+$ trytls https -- docker run --rm test-http-client-tls
```
Comment (jviide) [inline stubs/haskell-http-client-tls/README.md:34]: The image built above is called test-**haskell**-client-tls and here it's called test-**http**-client-tls. This is probably not intentional?

Comment (oherrala) [inline stubs/haskell-http-client-tls/README.md:34]: Fixed. Thanks!

**Comment by oherrala** [comment]
I think this is ready to be merged.

**Comment by Mamietti** [comment]
Works like a charm! The localhost test get kinda wonky though, how's that? Or is it OS X + Docker combo being weird...

**Comment by Mamietti** [comment]
ping @oherrala

**Comment by Mamietti** [comment]
Whoops, did not readme the README....

**Comment by jviide** [comment]
How does the localhost test get wonky? The end of the README file does mention a caveat concerning the dockerized tests:

> _Notice!_ Since trytls can't inject CA bundle inside container from outside, these tests will fail.

This will probably cause the localhost tests to fail.

**Pull request merged**